packages/react-instantsearch-core/src/core/utils.ts

```diff
@@ -62,3 +62,26 @@ export function addQueryID(hits, queryID) {
     __queryID: queryID,
   }));
 }
+
+/**
+ * Retrieve the value at a path of the object:
+ *
+ * @example
+ * getPropertyByPath(
+ *   { test: { this: { function: [{ now: { everyone: true } }] } } },
+ *   'test.this.function[0].now.everyone'
+ * ); // true
+ *
+ * getPropertyByPath(
+ *   { test: { this: { function: [{ now: { everyone: true } }] } } },
+ *   ['test', 'this', 'function', 0, 'now', 'everyone']
+ * ); // true
+ *
+ * @param object Source object to query
+ * @param path either an array of properties, or a string form of the properties, separated by .
+ */
+export const getPropertyByPath = (object: object, path: string[] | string) =>
+  (Array.isArray(path)
+    ? path
+    : path.replace(/\[(\d+)]/g, '.$1').split('.')
+  ).reduce((current, key) => (current ? current[key] : undefined), object);
```

Comment:
We loop over the full path even if the one of the member returns `undefined`. With recursion we might be able to avoid this unnecessary work. Worst case I'm fine with this implementation it's simple enough.

Reply:
True, however it’s very rare to see deep paths, so there’s no need to early exit here IMO